### ISSUE TITLE
Bot no longer connects to slack

### ISSUE THREAD

**Issue by UnbendableStraw** [rolfwessels/Command.Bot#17]
I think slack's changes are affecting the bot's ability to connect. They are deprecating legacy tokens while keeping existing ones still active, but for some reason mine no longer work. 

What happens is when I start the bot, the log sometimes says it connected but sometimes never says it connected. Regardless of the log output, the bot no longer appears online, and no longer responds or logs any direct mentions or messages. 

I tried setting up a bot App and using the oauth token but the bot logs a "not_allowed_token_type" when trying to use it.

Bot log with old legacy token and an attempt with the new oauth token

`2020-05-12 07:25:09,778 [1] INFO  [(null)] - Starting Command.Bot app
2020-05-12 07:25:09,825 [4] INFO  [(null)] - Start service
2020-05-12 07:25:09,825 [4] INFO  [(null)] - Starting service
2020-05-12 07:25:09,841 [4] INFO  [(null)] - Connecting to slack service
2020-05-12 07:25:09,935 [4] INFO  [(null)] - Services started
2020-05-12 07:26:10,671 [11] INFO  [(null)] - Stop service
2020-05-12 07:26:10,671 [11] INFO  [(null)] - Stop the service
2020-05-12 07:26:10,671 [11] INFO  [(null)] - Services stopped
2020-05-12 07:26:12,902 [1] INFO  [(null)] - Starting Command.Bot app
2020-05-12 07:26:12,949 [4] INFO  [(null)] - Start service
2020-05-12 07:26:12,949 [4] INFO  [(null)] - Starting service
2020-05-12 07:26:12,964 [4] INFO  [(null)] - Connecting to slack service
2020-05-12 07:26:13,058 [4] INFO  [(null)] - Services started
2020-05-12 07:26:14,512 [5] INFO  [(null)] - Connected
2020-05-12 07:51:09,628 [5] INFO  [(null)] - Stop service
2020-05-12 07:51:09,628 [5] INFO  [(null)] - Stop the service
2020-05-12 07:51:09,628 [5] INFO  [(null)] - Services stopped
2020-05-12 07:52:22,654 [1] INFO  [(null)] - Starting Command.Bot app
2020-05-12 07:52:22,701 [4] INFO  [(null)] - Start service
2020-05-12 07:52:22,701 [4] INFO  [(null)] - Starting service
2020-05-12 07:52:22,717 [4] INFO  [(null)] - Connecting to slack service
2020-05-12 07:52:22,811 [4] INFO  [(null)] - Services started
2020-05-12 07:52:23,217 [5] ERROR [(null)] - One or more errors occurred.
System.AggregateException: One or more errors occurred. ---> SlackConnector.Exceptions.CommunicationException: Error occured while posting message 'not_allowed_token_type'
   at SlackConnector.Connections.Clients.ResponseVerifier.VerifyResponse(StandardResponse response)
   at SlackConnector.Connections.Clients.Handshake.FlurlHandshakeClient.<FirmShake>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SlackConnector.SlackConnector.<Connect>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Command.Bot.Core.SlackService.<Connect>d__6.MoveNext()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) SlackConnector.Exceptions.CommunicationException: Error occured while posting message 'not_allowed_token_type'
   at SlackConnector.Connections.Clients.ResponseVerifier.VerifyResponse(StandardResponse response)
   at SlackConnector.Connections.Clients.Handshake.FlurlHandshakeClient.<FirmShake>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SlackConnector.SlackConnector.<Connect>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Command.Bot.Core.SlackService.<Connect>d__6.MoveNext()<---

2020-05-12 07:52:47,460 [10] INFO  [(null)] - Stop service
2020-05-12 07:52:47,460 [10] INFO  [(null)] - Stop the service
2020-05-12 07:52:47,460 [10] INFO  [(null)] - Services stopped

`

**Comment by rolfwessels** [comment]
👍 I was not aware that they changed token types.

I will have to spend some time to see what the new auth scheme is like.

Thanks for the feedback.

**Comment by UnbendableStraw** [comment]
I spent some time with Slack support trying to see if we could get it working again to no avail, we tried regenerating a new legacy token, then we tried a classic bot app token, then lastly tried a new bot app token which I didnt expect to work because of the new oauth....but I really don't know much about this : (

Thank you for your work! I'll stay tuned

**Comment by rolfwessels** [comment]
Hi, I think I found the problem. Connection with tokens starting with 'xoxb-' should still work.

Could you check out version 1.0.4 and let me know how it goes?

**Comment by UnbendableStraw** [comment]
Sure, so it looks like if I try with the xoxb token from a classic app the app the bot gives some sort of error on startup (first picture attached)

If I try with a "xoxp" legacy token, the bot says it connects but it doesnt appear online on slack and doesnt respond (second pic)

If I try with a "xoxb" bot token from a regular slack app, the bot says it connects, bot appears online on slack, but doesn't respond to any slack messages

<img width="415" alt="Screen Shot 2020-05-16 at 9 10 33 PM" src="https://user-images.githubusercontent.com/22043907/82135729-e763a580-97ba-11ea-919c-2054fa8c82d7.png">
<img width="413" alt="Screen Shot 2020-05-16 at 9 10 50 PM" src="https://user-images.githubusercontent.com/22043907/82135732-ed598680-97ba-11ea-8f34-943b9f799844.png">

**Comment by rolfwessels** [comment]
I'm not getting the same result:

I created a new slack account. Went to app bot
![image](https://user-images.githubusercontent.com/643179/82139662-e027b080-9829-11ea-987b-627ad037d3bf.png)


Created a new bot:
![image](https://user-images.githubusercontent.com/643179/82139683-09484100-982a-11ea-8ce5-1febe0def3d4.png)

Then I ran:

![image](https://user-images.githubusercontent.com/643179/82139696-254be280-982a-11ea-9b64-8304787871db.png)

Now the bot is responding as expected:

![image](https://user-images.githubusercontent.com/643179/82139716-4280b100-982a-11ea-8ab1-8fe4f406f590.png)

The second command looks like its connecting. Just not sure why you are not getting any responses back.
